packages/seda-chain-contracts/src/utils.rs

```diff
@@ -24,3 +25,42 @@ pub fn check_eligibility(deps: &DepsMut, dr_executor: Addr) -> Result<bool, Cont
     let is_eligible = ELIGIBLE_DATA_REQUEST_EXECUTORS.load(deps.storage, dr_executor)?;
     Ok(is_eligible)
 }
+
+pub fn get_attached_funds(funds: &[Coin], token: String) -> Result<u128, ContractError> {
+    let amount: Option<u128> = funds
+        .iter()
+        .find(|coin| coin.denom == token)
+        .map(|coin| coin.amount.u128());
+    amount.ok_or(ContractError::NoFunds)
+}
+
+pub fn pad_to_32_bytes(value: u128) -> [u8; 32] {
```

Comment:
Looks like since you ref value here you can change to pass by ref. `value: &u128`.